osu.Game/Graphics/UserInterface/Counter.cs

```diff
@@ -0,0 +1,110 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public abstract class Count : AutoSizeContainer
+    {
+        private Sprite buttonSprite;
+        private Sprite glowSprite;
+        private SpriteText keySpriteText;
+        private SpriteText countSpriteText;
+
+        public override string Name { get; }
+        public KeyCounter ParentCounter { get; set; }

```

Comment:
Don't ever access parents. We are aiming to never do this.

You'd be better off maintaining a list of keys at the KeyCounter and setting a property on them instead.